job/server/src/main/java/alluxio/master/job/JobCoordinator.java

```diff
@@ -153,10 +154,12 @@ public synchronized void cancel() {
    * @param taskInfoList List of @TaskInfo instances to update
    */
   public synchronized void updateTasks(List<TaskInfo> taskInfoList) {
-    for (TaskInfo taskInfo : taskInfoList) {
-      mJobInfo.setTaskInfo(taskInfo.getTaskId(), taskInfo);
+    synchronized (mJobInfo) {
```

Comment:
this and following synchronization added is redundant? the previous way of synchronization is not fine grained but should be functionally correct by synchronizing on the JobCoordinator instance?